src/components/CardsContainer.tsx

```diff
@@ -27,57 +44,50 @@ export default function CardsContainer() {
-	// POINT_CODES.JOIN = user joined room and hasn't participated yet.
-	// POINT_CODES.RESET = user has participated at least once and had points cleared.
 	const makeStringPoints = (message: number) => {
 		if (message === POINT_CODES.JOIN || message === POINT_CODES.RESET) return '--'
-		if (POINT_CODES.QUESTION) return '?'
+		if (message === POINT_CODES.QUESTION) return '?'
 		return message.toString()
 	}
 
 	const numberOfBlankCards = usersPointsData.filter(
 		({ message }) => message === POINT_CODES.JOIN || message === POINT_CODES.RESET,
 	).length
 
+	const usersPointsForCards = showStoryPoints ? sortedUsersPoints : usersPointsData
+
 	// console.log('%c>>> CardsContainer: allUsersStoryPoints', 'color: #f70', allUsersStoryPoints)
 	// console.log('%c>>> CardsContainer: usersPointsData', 'color: red', usersPointsData)
 	// console.log('%c>>> CC socketListener: sortedUsersPoints', 'color: #5f0', sortedUsersPoints)
 	// console.log('%c>>> CardsContainer: showStoryPoints', 'color: #f60', showStoryPoints)
 
-	// TODO create a combo UserCard with all needed props, calculations and JSX. Modify the `key` to be different for each of the 2 arrays of points.
-	// key={timeStamp.toString() + showStoryPoints.toString()}
+	// NOTE: the key needs to change when showStoryPoints changes and therefore
+	// the array changes to sortedUsersPoints. The timestamp makes each key unique, and
+	// showStoryPoints causes the key to change when the array changes. So:
+	// key={`${timeStamp.toString()}-${showStoryPoints.toString()}`}
 	return (
-		<div className='relative pt-2 pb-56 flex justify-center items-center flex-wrap text-center gap-8 text-gray-300 border-0 border-red-800 w-full'>
-			{showStoryPoints
-				? sortedUsersPoints.map(({ message, userName, imageNumber, timeStamp }, index, array) => {
-						const storyPoint = makeStringPoints(message as number)
-						return (
-							<UserShowPointsCard
-								key={timeStamp}
-								name={userName}
-								storyPoint={storyPoint}
-								imageNumber={imageNumber}
-								index={index}
-								numberOfCards={array.length}
-								numberOfBlanks={numberOfBlankCards}
-							/>
-						)
-					})
-				: usersPointsData.map(({ message, userName, imageNumber, timeStamp }) => {
-						const storyPoint = makeStringPoints(message as number)
-						return (
-							<UserCard
-								key={timeStamp}
-								name={userName}
-								storyPoint={storyPoint}
-								imageNumber={imageNumber}
-								showPoints={showStoryPoints}
-							/>
-						)
-					})}
+		<div
+			ref={containerRef}
+			className='relative pt-2 pb-20 flex justify-center items-center flex-wrap text-center gap-8 text-gray-300 border-0 border-red-800 w-full'
+		>
+			{usersPointsForCards.map(({ message, userName, imageNumber, timeStamp }, index, array) => {
+				const storyPoint = makeStringPoints(message as number)
+				return (
+					<UserPointsCard
+						key={`${timeStamp.toString()}-${showStoryPoints.toString()}`}
```

Comment:
suggestion: Using the `key` prop is uncommon, so it's important to leave a code comment explaining why it is needed in this case :)

Reply:
Good idea. I'll add that.

Reply:
DONE